sdk/servicebus/Azure.Messaging.ServiceBus/tests/Infrastructure/ServiceBusScope.cs

```diff
@@ -0,0 +1,627 @@
+                    // does not help to ensure resource cleanup only flags the test itself as a failure.
+                    //
+                    // If a queue fails to be deleted, removing of the associated namespace at the end of the
+                    // test run will also remove the orphan.
+                }
+
+                _disposed = true;
+            }
+        }
+
+        /// <summary>
+        ///   Provides access to dynamically created Service Bus Topic instance which exists only in the context
+        ///   of the scope; disposal removes the resource from Azure.
+        /// </summary>
+        ///
+        /// <seealso cref="IAsyncDisposable" />
+        ///
+        public sealed class TopicScope : IAsyncDisposable
+        {
+            /// <summary>Serves as a sentinel flag to denote when the instance has been disposed.</summary>
+            private bool _disposed = false;
+
+            /// <summary>
+            ///   The name of the Service Bus namespace associated with the queue.
+            /// </summary>
+            ///
+            public string NamespaceName { get; }
+
+            /// <summary>
+            ///  The name of the topic.
+            /// </summary>
+            ///
+            public string TopicName { get; }
+
+            /// <summary>
+            ///   The set of names for the subscriptions associated with the topic.
+            /// </summary>
+            ///
+            public IReadOnlyList<string> SubscriptionNames { get; }
+
+            /// <summary>
+            ///   A flag indicating if the topic should be removed when the scope is complete.
+            /// </summary>
+            ///
+            /// <value><c>true</c> if the queue was should be removed at scope completion; otherwise, <c>false</c>.</value>
+            ///
+            private bool ShouldRemoveAtScopeCompletion { get; }
+
+            /// <summary>
+            ///   Initializes a new instance of the <see cref="TopicScope"/> class.
+            /// </summary>
+            ///
+            /// <param name="serviceBusNamespaceName">The name of the Service Bus namespace to which the queue is associated.</param>
+            /// <param name="topicName">The name of the topic.</param>
+            /// <param name="subscriptionNames">The set of names for the subscriptions </param>
+            /// <param name="shouldRemoveAtScopeCompletion">A flag indicating whether the topic should be removed when the scope is complete.</param>
+            ///
+            public TopicScope(string serviceBusNamespaceName,
+                              string topicName,
+                              IReadOnlyList<string> subscriptionNames,
+                              bool shouldRemoveAtScopeCompletion)
+            {
+                NamespaceName = serviceBusNamespaceName;
+                TopicName = topicName;
+                SubscriptionNames = subscriptionNames;
```

Comment:
Hmm.. is this used anywhere?

Reply:
The constructor?   Yes, whenever you call `CreateWithQueue` on the `ServiceBusScope`.

Reply:
Sorry I meant the SubscriptionNames property.

Reply:
If you create a topic scope and want to have more than one subscription created or want to specify an exact name, you'll pass them in.    The majority of track one tests used a single subscription with the default name, but there were a small set of tests that asked for multiple subscriptions to be created.    Since we're likely to be mirroring many of their scenarios, I included the functionality.

Reply:
I didn't see anywhere that is actually reading that property though.

Reply:
I don't think that I saw any tests that were hooking into subscriptions for topics just yet.   The names will be needed when creating a `SubscriptionClient` or the equivalent concept.    _([example](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Microsoft.Azure.ServiceBus/tests/TopicClientTests.cs#L31))_